packages/buidler-core/test/internal/solidity/compilationGroup.ts

```diff
@@ -0,0 +1,1225 @@
+import { assert } from "chai";
+import * as fs from "fs";
+import path from "path";
+
+import { SolidityFilesCache } from "../../../src/builtin-tasks/utils/solidity-files-cache";
+import { createCompilationGroups } from "../../../src/internal/solidity/compilationGroup";
+import { DependencyGraph } from "../../../src/internal/solidity/dependencyGraph";
+import {
+  ResolvedFile,
+  Resolver,
+} from "../../../src/internal/solidity/resolver";
+
+const defaultOptimizer = {
+  enabled: false,
+  runs: 200,
+};
+
+const solcConfig055 = {
+  compilers: [{ version: "0.5.5", optimizer: defaultOptimizer }],
+};
+const solcConfig066 = {
+  compilers: [{ version: "0.6.6", optimizer: defaultOptimizer }],
+};
+const solcConfig055and066 = {
+  compilers: [
+    { version: "0.5.5", optimizer: defaultOptimizer },
+    { version: "0.6.6", optimizer: defaultOptimizer },
+  ],
+};
+
+const projectRoot = fs.realpathSync(".");
+
+class MockFile {
+  public readonly globalName: string;
+  public readonly absolutePath: string;
+
+  constructor(
+    public name: string,
+    public versionPragmas: string[],
+    public readonly modified: "new" | "modified" | "not-modified"
+  ) {
+    this.globalName = `contracts/${name}.sol`;
+    this.absolutePath = path.join(projectRoot, "contracts", `${name}.sol`);
+  }
+}
+
+async function createMockData(
```

Comment:
This function is a monstrosity, but I wanted the tests to be as readable as possible, so I moved all the setup complexity here.